Got a friend to do it for me on a Xiaomi Vacuum 1st gen
```
Running command fan_speed_presets
Traceback (most recent call last):
  File "/home/pi/temp_venv/bin/miiocli", line 11, in <module>
    load_entry_point('python-miio', 'console_scripts', 'miiocli')()
  File "/home/pi/temp_venv/src/python-miio/miio/cli.py", line 44, in create_cli
    return cli(auto_envvar_prefix="MIIO")
  File "/home/pi/temp_venv/src/python-miio/miio/click_common.py", line 59, in _call_
    return self.main(*args, **kwargs)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/pi/temp_venv/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/pi/temp_venv/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/pi/temp_venv/src/python-miio/miio/click_common.py", line 280, in wrap
    kwargs["result"] = func(*args, **kwargs)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/click/decorators.py", line 73, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/pi/temp_venv/src/python-miio/miio/click_common.py", line 245, in command_callback
    return miio_command.call(miio_device, *args, **kwargs)
  File "/home/pi/temp_venv/src/python-miio/miio/click_common.py", line 193, in call
    return method(*args, **kwargs)
  File "/home/pi/temp_venv/src/python-miio/miio/vacuum.py", line 476, in fan_speed_presets
    self._autodetect_model()
  File "/home/pi/temp_venv/src/python-miio/miio/vacuum.py", line 445, in _autodetect_model
    info = self.info()
  File "/home/pi/temp_venv/src/python-miio/miio/device.py", line 156, in info
    return DeviceInfo(self._protocol.send("miIO.info"))
  File "/home/pi/temp_venv/src/python-miio/miio/miioprotocol.py", line 158, in send
    m = Message.build(msg, token=self.token)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/construct/core.py", line 336, in build
    self.build_stream(obj, stream, **contextkw)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/construct/core.py", line 348, in build_stream
    self._build(obj, stream, context, "(building)")
  File "/home/pi/temp_venv/lib/python3.7/site-packages/construct/core.py", line 2005, in _build
    buildret = sc._build(subobj, stream, context, path)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/construct/core.py", line 2443, in _build
    return self.subcon._build(obj, stream, context, path)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/construct/core.py", line 3996, in _build
    buildret = self.subcon._build(obj, stream, context, path)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/construct/core.py", line 4283, in _build
    buildret = self.subcon._build(value, stream, context, path)
  File "/home/pi/temp_venv/lib/python3.7/site-packages/construct/core.py", line 701, in _build
    obj2 = self._encode(obj, context, path)
  File "/home/pi/temp_venv/src/python-miio/miio/protocol.py", line 157, in _encode
    json.dumps(obj).encode("utf-8") + b"\x00", context["_"]["token"]
  File "/home/pi/temp_venv/src/python-miio/miio/protocol.py", line 83, in encrypt
    cipher = Cipher(algorithms.AES(key), modes.CBC(iv), backend=default_backend())
  File "/home/pi/temp_venv/lib/python3.7/site-packages/cryptography/hazmat/backends/_init_.py", line 15, in default_backend
    from cryptography.hazmat.backends.openssl.backend import backend
  File "/home/pi/temp_venv/lib/python3.7/site-packages/cryptography/hazmat/backends/openssl/_init_.py", line 7, in <module>
    from cryptography.hazmat.backends.openssl.backend import backend
  File "/home/pi/temp_venv/lib/python3.7/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 75, in <module>
    from cryptography.hazmat.bindings.openssl import binding
  File "/home/pi/temp_venv/lib/python3.7/site-packages/cryptography/hazmat/bindings/openssl/binding.py", line 15, in <module>
    from cryptography.hazmat.bindings._openssl import ffi, lib
ImportError: /usr/lib/arm-linux-gnueabihf/libssl.so.1.1: version `OPENSSL_1_1_1' not found (required by /home/pi/temp_venv/lib/python3.7/site-packages/cryptography/hazmat/bindings/_openssl.abi3.so)
```

Didn't go that well..

--- a/homeassistant/components/xiaomi_miio/manifest.json
+++ b/homeassistant/components/xiaomi_miio/manifest.json
@@ -2,7 +2,7 @@
   "domain": "xiaomi_miio",
   "name": "Xiaomi miio",
   "documentation": "https://www.home-assistant.io/integrations/xiaomi_miio",
-  "requirements": ["construct==2.9.45", "python-miio==0.4.8"],
+  "requirements": ["construct==2.9.45", "python-miio==0.5.0"],
   "dependencies": [],
   "codeowners": ["@rytilahti", "@syssi"]
 }

--- a/homeassistant/components/xiaomi_miio/vacuum.py
+++ b/homeassistant/components/xiaomi_miio/vacuum.py
@@ -281,7 +281,9 @@ class MiroboVacuum(StateVacuumDevice):
         if self.vacuum_state is not None:
             speed = self.vacuum_state.fanspeed
             if speed in self._fan_speeds.values():
-                return [key for key, value in self._fan_speeds.items() if value == speed][0]
+                return [
+                    key for key, value in self._fan_speeds.items() if value == speed
+                ][0]
             return speed
 
     @property

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -1623,7 +1623,7 @@ python-juicenet==0.1.6
 # python-lirc==1.2.3
 
 # homeassistant.components.xiaomi_miio
-python-miio==0.4.8
+python-miio==0.5.0
 
 # homeassistant.components.mpd
 python-mpd2==1.0.0

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -581,7 +581,7 @@ python-forecastio==1.4.0
 python-izone==1.1.2
 
 # homeassistant.components.xiaomi_miio
-python-miio==0.4.8
+python-miio==0.5.0
 
 # homeassistant.components.nest
 python-nest==4.1.0

